west.yml : update modules/tee/tf-m/trusted-firmware-m

FRASTM opened the pull request:
Takes the PR
https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/115 to fix the build error on stm32u5 due to gcc revision

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/80932

tomi-font reviewed (changes requested):
You'll need to rebase your PR in the TF-M repo, it's missing some changes which is provoking the CI failures here.

FRASTM commented:
rebase on 44d5d8aef23f5ace319fd1cdb13d15aa86bc2bed

tomi-font commented:
You haven't rebased your [TF-M PR](https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/115), you have only rebased your Zephyr one. I wouldn't be surprised CI still fails.

tomi-font commented:
```
/__w/zephyr/modules/tee/tf-m/trusted-firmware-m/platform/ext/target/stm/common/stm32u5xx/secure/tfm_hal_isolation.c: In function 'tfm_hal_bind_boundary':
/__w/zephyr/modules/tee/tf-m/trusted-firmware-m/platform/ext/target/stm/common/stm32u5xx/secure/tfm_hal_isolation.c:364:29: warning: unused variable 'plat_data_ptr' [-Wunused-variable]
  364 |     struct platform_data_t *plat_data_ptr;
      |                             ^~~~~~~~~~~~~
```
https://github.com/zephyrproject-rtos/zephyr/pull/82534/checks?check_run_id=35764793910
:thinking:

erwango commented:
@tomi-font According to CI, https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/115 could be merged, if you're up to.

FRASTM committed:
west.yml : update modules/tee/tf-m/trusted-firmware-m

Updates the https://github.com/zephyrproject-rtos/trusted-firmware-m/
to fix the build error on stm32u5 due to gcc revision

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
- update the SHA1 of the modules/tee/tf-m/trusted-firmware-m
after merge of https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/115

- DNM removed

tomi-font reviewed (approved):
For future reference, no need to do it now (to avoid yet another CI churn): the commit would typically rather go something like `modules: tf-m: fix build on stm32u5 with GCC 11/12`.
